xarray/core/dataset.py

```diff
@@ -1849,7 +1849,7 @@ def to_zarr(
 
         See Also
         --------
-        :doc:`io.zarr`
+        :ref:`io.zarr`
```

Comment:
Cool, that works!

![image](https://user-images.githubusercontent.com/15079414/115098681-57186e00-9eff-11eb-9951-a786cae0200d.png)

I suppose the link text could be more informative, but also it seems fine this way.

Reply:
yeah i'm not sure what is the best way to link to other parts of the docs from the docstring. I see "See also" is often used for linking to other API functions

Reply:
other option although tedious and probably not worth is to rename that section as zarr-io as everything in http://xarray.pydata.org/en/stable/io.html# would have to has a suffix of -io.
But it this as good as a way to keep http://xarray.pydata.org/en/stable/io.html#zarr and http://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_zarr.html#xarray.Dataset.to_zarr linked

Reply:
To make it slightly more informative, it could be something like ``:ref:`Reading and writing Zarr files <io.zarr>` ``.

Yeah, in numpydoc *See Also* section is usually used to link to other parts of the code. And in the xarray docs there doesn't seem to be much of a precedent for using it to link to the user guide. But it does work as we can see. It might be cool to do similar for other API docs.

Reply:
> it could be something like :ref:`Reading and writing Zarr files <io.zarr>`

either that, or something like
```suggestion
        :ref:`io.zarr`
            Link to the I/O user-guide, with more details and examples.
```
(or both), which would help when viewing with `pydoc` / reading the source code.

Reply:
@keewis It seems that it doesn't work with the description, which would be after `:`, indented or on same line. Also, including the `:ref:`, even without description, seems to break the parsing of *See Also* references to API, which is the normal usage. Maybe instead, a new Sphinx directive to list relevant user guide sections could be created.

Reply:
that's curious. I tried this locally and it seemed to work: 
![screenshot](https://user-images.githubusercontent.com/14808389/115156331-04b48a00-a084-11eb-9bc1-dcdc3e3662b8.png)

You don't need the colon in _See Also_ if you put the description on a new line and indent it

Reply:
Ah, interesting, that looks good. I'll have to check what was different in my case where I was trying it.